File: local-data-persistent/LatihanRepository (starter)/java/app/src/main/java/com/dicoding/newsapp/utils/DateFormatter.java
package com.dicoding.newsapp.utils;

import androidx.annotation.Nullable;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class DateFormatter {
    @Nullable
    public static String formatDate(String currentDate) {
        String currentFormat = "yyyy-MM-dd'T'hh:mm:ss'Z'";
        String targetFormat = "dd MMM yyyy | HH:mm";
        String timezone = "GMT";
        DateFormat currentDf = new SimpleDateFormat(currentFormat, Locale.getDefault());
        currentDf.setTimeZone(TimeZone.getTimeZone(timezone));
        DateFormat targetDf = new SimpleDateFormat(targetFormat, Locale.getDefault());
        String targetDate = null;
        try {
            Date date = currentDf.parse(currentDate);
            if (date != null) {
                targetDate = targetDf.format(date);
            }
        } catch (ParseException ex) {
            ex.printStackTrace();
        }
        return targetDate;
    }
}

File: local-data-persistent/LatihanRepository (starter)/kotlin/app/src/main/java/com/dicoding/newsapp/utils/DateFormatter.kt
package com.dicoding.newsapp.utils

import java.text.DateFormat
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateFormatter {
    fun formatDate(currentDate: String): String? {
        val currentFormat = "yyyy-MM-dd'T'hh:mm:ss'Z'"
        val targetFormat = "dd MMM yyyy | HH:mm"
        val timezone = "GMT"
        val currentDf: DateFormat = SimpleDateFormat(currentFormat, Locale.getDefault())
        currentDf.timeZone = TimeZone.getTimeZone(timezone)
        val targetDf: DateFormat = SimpleDateFormat(targetFormat, Locale.getDefault())
        var targetDate: String? = null
        try {
            val date = currentDf.parse(currentDate)
            if (date != null) {
                targetDate = targetDf.format(date)
            }
        } catch (ex: ParseException) {
            ex.printStackTrace()
        }
        return targetDate
    }
}
